rpcclient/cookiefile.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2017 The Namecoin developers
+// Copyright (c) 2019 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package rpcclient
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+)
+
+func readCookieFile(path string) (username, password string, err error) {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return
+	}
+
+	s := strings.TrimSpace(string(b))
```

Comment:
Considering Core uses base64 encoding for their client cookie authentication, those libraries could be used instead of a binary => string cast

Reply:
You're talking about https://github.com/bitcoin/bitcoin/blob/f4a0d27e85754d60804ffa36e415b67c263180b9/src/bitcoin-cli.cpp#L345 , right?  AFAICT from looking at the Bitcoin Core source code, while it's true that the Core cookie reader doesn't try to parse out a separate username and password (and instead simply submits the line extracted from the cookie file verbatim), the code that actually creates the cookie file does distinguish between the username and password components.

I think it's generally a good thing to try to parse out the username and password separately, for 2 reasons:

1. It simplifies the integration with existing code that expects authentication to use a username+password rather than a combined blob.
2. It will better detect cases where the cookie file doesn't look like a standard cookie file, which are probably an indication of some kind of error.

Reply:
That makes sense, I'm cool with the user/pass separation. I still have an issue with this line though - why are the spaces being trimmed? I just added a space in a cookie file with Core and it worked.
Parsing only the first line of the file makes sense, and it is what core does in https://github.com/bitcoin/bitcoin/blob/42d0eca725a83c999e2b67e33dfc7bcc96288dc3/src/rpc/request.cpp#L108, but spaces in general seem a little arbitrary

Reply:
Replaced `TrimSpace` with reading the first line.